packages/db/src/executions.ts

```diff
@@ -0,0 +1,71 @@
+import Knex from 'knex';
+
+import { ExecutionRecord } from '@cumulus/types/api/executions';
+import { PostgresExecution } from './types';
+// TODO move to common location
+const {
+  getParentExecutionCumulusId,
+  getAsyncOperationCumulusId,
+  getCollectionCumulusId,
+} = require('../../api/lambdas/sf-event-sqs-to-db-records/utils');
+
+/**
+ * Translate execution record from Dynamo to RDS.
+ *
+ * @param {AWS.DynamoDB.DocumentClient.AttributeMap} dynamoRecord
+ *   Source record from DynamoDB
+ * @param {AWS.DynamoDB.DocumentClient.AttributeMap} knex
+ *   Knex client
+ * @returns {PostgresExecutionRecord} - converted Execution
+ */
+export const translateApiExecutionToPostgresExecution = async (
+  dynamoRecord: ExecutionRecord,
+  knex: Knex
+): Promise<PostgresExecution> => {
+  // Map old record to new schema.
+  const translatedRecord: PostgresExecution = {
+    async_operation_cumulus_id: (
+      dynamoRecord.asyncOperationId ? await getAsyncOperationCumulusId(
+        dynamoRecord.asyncOperationId, knex
+      ) : undefined
+    ),
+    status: dynamoRecord.status,
+    tasks: JSON.stringify(dynamoRecord.tasks),
+    error: JSON.stringify(dynamoRecord.error),
+    arn: dynamoRecord.arn,
+    duration: dynamoRecord.duration,
+    original_payload: JSON.stringify(dynamoRecord.originalPayload),
+    final_payload: JSON.stringify(dynamoRecord.finalPayload),
+    workflow_name: dynamoRecord.name,
+  };
+
+  if (dynamoRecord.timestamp) {
+    translatedRecord.timestamp = new Date(Number(dynamoRecord.timestamp));
+  }
+
+  if (dynamoRecord.collectionId) {
+    // TODO is there a helper for this?
```

Comment:
I didn't see anything that translates from the `name___version` format of a collection ID to either an array of those values or a cumulus_id directly. Again, this might be a good small thing to add to the new Collection model unless it exists somewhere already and I'm not seeing it?

Reply:
Duplicate comment